### PR TITLE
psql network settings and port range support

### PR DESCRIPTION
**Pull request by aelesbao**

### CODE REVIEW

File: sql/src/main/java/io/crate/protocols/postgres/PostgresNetty.java [hunk 1]
```diff
@@ -22,36 +22,66 @@
...
+    private volatile List<Channel> serverChannels = new ArrayList<>();
 
     @Inject
-    public PostgresNetty(Settings settings, SQLOperations sqlOperations) {
+    public PostgresNetty(Settings settings, SQLOperations sqlOperations, NetworkService networkService) {
         super(settings);
         this.sqlOperations = sqlOperations;
+        this.networkService = networkService;
+
+        port = settings.get("psql.netty.port", settings.get("psql.port"));
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/protocols/postgres/PostgresNetty.java]: why have two variants of setting names?

File: sql/src/main/java/io/crate/protocols/postgres/PostgresNetty.java [hunk 2]
```diff
@@ -96,6 +225,9 @@ protected void doStop() {
 
     @Override
     protected void doClose() {
+    }
 
+    public BoundTransportAddress boundAddress() {
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/protocols/postgres/PostgresNetty.java]: why do you expose the boundAddress? It's not used anywhere?

File: sql/src/main/java/io/crate/protocols/postgres/PostgresNetty.java (hunk 1 above)
Comment (mfussenegger) [inline sql/src/main/java/io/crate/protocols/postgres/PostgresNetty.java]: I think we should also register these settings as `CrateSettings` and you can use the `Int/String-Setting`  classes

File: sql/src/main/java/io/crate/protocols/postgres/PostgresNetty.java [hunk 3]
```diff
@@ -75,11 +105,110 @@ public ChannelPipeline getPipeline() throws Exception {
...
+        boundAddress = createBoundAddress();
+    }
+
+    private BoundTransportAddress createBoundAddress() {
+        // Bind and start to accept incoming connections.
+        InetAddress hostAddresses[];
+        try {
+            hostAddresses = networkService.resolveBindHostAddresses(bindHosts);
+        } catch (IOException e) {
+            throw new BindPostgresException("Failed to resolve host [" + Arrays.toString(bindHosts) + "]", e);
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/protocols/postgres/PostgresNetty.java]: What kind of exceptions would we get if we didn't wrap them?

**Comment by mfussenegger** [comment]
Can you also update the configurations documentation?

File: sql/src/main/java/io/crate/protocols/postgres/PostgresNetty.java (hunk 3 above)
Comment (aelesbao) [inline sql/src/main/java/io/crate/protocols/postgres/PostgresNetty.java]: mostly `SocketException`, `UnknownHostException` and other IO errors that may occur when resolving the address

**Comment by aelesbao** [comment]
@mfussenegger could you please check again?

File: blackbox/docs/configuration.txt [hunk 4]
```diff
@@ -403,6 +403,17 @@ Ports
...
+.. _psql.port:
+
+**psql.port**
+  | *Runtime:*   ``no``
+
+  This defines the TCP port range to which the Crate Postgres service will be bound to.
+  It defaults to ``4250-4300``. Always the first free port in this range is used. If this is set
+  to an integer value it is considered as an explicit single port.
+
+  The PostgreSQL wire protocol is used by the Java client.
```
Comment (mfussenegger) [inline blackbox/docs/configuration.txt]: please remove this sentence

**Comment by mfussenegger** [comment]
lgtm otherwise

**Pull request merged**